flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -775,7 +775,7 @@ class ExprLowering {
       auto chTy =
           converter.genType(Fortran::common::TypeCategory::Character, KIND);
       auto arrayTy = fir::SequenceType::get(shape, chTy);
-      auto idxTy = builder.getIndexType();
+      auto idxTy = builder.getIntegerType(32);
```

Comment:
What is the motivation for changing this index type?  (There is another instance of this in CharacterExpr.cpp.)

Reply:
The lowering through to LLVM doesn't like 64 bit ints. I think this fix may be obsolete already though.